### PR TITLE
Transfer tokens with Mask

### PR DESCRIPTION
**Pull request by guanbinrui**
+ [x] Add the transfer dialog UI.
+ [x] Merge the share dialog UI as a sub-tab of the transfer dialog UI.

test-1
test-2

### CODE REVIEW

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    const [address, setAddress] = useState('')
+    const [memo, setMemo] = useState('')
+
+    // balance
+    const { value: tokenBalance = '0' } = useTokenBalance(token?.type ?? EthereumTokenType.Ether, token?.address ?? '')
+
+    const onChangeAmount = useCallback((ev: ChangeEvent<HTMLInputElement>) => {
+        const _amount = ev.currentTarget.value
+        if (_amount === '') setAmount('')
+        if (/^\d+[\.]?\d*$/.test(_amount)) setAmount(_amount)
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: 🤔Why not parse with Number.parseFloat then check if NaN?

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+                                </IconButton>
+                            </InputAdornment>
+                        ),
+                    }}
+                    variant="outlined"
+                />
+            </form>
+            <Box
+                className={classes.qr}
+                sx={{
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:233]: I strongly not suggest you mix using `makeStyles` and `Box`. MUI is migrating its styling engine to emotion. Now `Box` is powered by emotion and `makeStyles` is powered by JSS. If you mix them and you need to overwrite some property, you may encounter CSS order problem

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx (hunk 1 above)
Comment (guanbinrui) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: It prohibits users from entering non-numeric content.

File: packages/maskbook/src/web3/hooks/useERC721TokenTransferCallback.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function useERC721TokenTransferCallback(address: string, id?: string, recipient?: string) {
+    // todo
```
Comment (Jack-Works) [inline packages/maskbook/src/web3/hooks/useERC721TokenTransferCallback.ts:2]: remove this file?

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx (hunk 2 above)
Comment (guanbinrui) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:233]: @lelenei Take a look.

File: packages/maskbook/src/web3/hooks/useERC721TokenTransferCallback.ts (hunk 3 above)
Comment (guanbinrui) [reply packages/maskbook/src/web3/hooks/useERC721TokenTransferCallback.ts:2]: Implemented a simple ver.

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx (hunk 1 above)
Comment (Jack-Works) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: another question, why not use the number type input box?

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx (hunk 2 above)
Comment (lelenei) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:233]: ok

File: packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx (hunk 1 above)
Comment (lelenei) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: In this way, i can clearly input 0

Comment (guanbinrui) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: Cannot get value in this way `inputNode.value` if we use a number type input. @lelenei Take a look at `<TokenAmountPanel />` component, a bunch of helpful properties can be added to this component.

```jsx
inputProps: {
      autoComplete: 'off',
      autoCorrect: 'off',
      title: 'Token Amount',
      inputMode: 'decimal',
      min: 0,
      minLength: MIN_AMOUNT_LENGTH,
      maxLength: MAX_AMOUNT_LENGTH,
      pattern: '^[0-9]*[.,]?[0-9]*$',
      placeholder: '0.0',
      spellCheck: false,
}
```

Comment (lelenei) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: TokenAmoumntPanel type is text

Comment (lelenei) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TransferDialog.tsx:73]: [
image
](url)

**Pull request merged**